src/plugin/modules/widgets/JobsWidget.js

```diff
@@ -0,0 +1,185 @@
+define([
+    'jquery',
+    'bluebird',
+    './DashboardWidget',
+    'kb_common/jsonRpc/dynamicServiceClient',
+    'kb_common/dynamicTable',
+], function($, Promise, DashboardWidget, DynamicService, DynamicTable) {
+    'use strict';
+    return Object.create(DashboardWidget, {
+
+        numHours : { value : 48 },
+
+        init: {
+            value: function(cfg) {
+                cfg.name = 'JobsWidget';
+                cfg.title = 'Jobs run in last ' + this.numHours + ' hours';
+                this.DashboardWidget_init(cfg);
+
+                return this;
+            }
+        },
+        setup: {
+            value: function() {
+                return this;
+            }
+        },
+
+        /* takes a timestamp and turns it into a locale string. If no timestamp is present, then it'll put in ...
+           atm, the only place this should happen is the finish_time on an unfinished job. */
+        reformatDate : { value : function(date) {
+          var timestamp = parseInt(date, 10);
+          if (Number.isNaN(timestamp)) {
+            return '...';
+          }
+          else {
+            return new Date(timestamp).toLocaleString();
+          }
+        }},
+
+        reformatInterval : { value : function(interval) {
+          var timestamp = parseInt(interval, 10) / 1000;
+          if (Number.isNaN(timestamp)) {
+            return '...';
+          }
+          else {
+            return this.getNiceDuration(timestamp);
+          }
+        }},
+
+        getNiceDuration: { value : function (seconds) {
+          var hours = Math.floor(seconds / 3600);
+          seconds = seconds - (hours * 3600);
+          var minutes = Math.floor(seconds / 60);
+          seconds = seconds - (minutes * 60);
+
+          var duration = '';
+          if (hours > 0) {
+              duration = hours + 'h ' + minutes + 'm';
+          } else if (minutes > 0) {
+              duration = minutes + 'm ' + Math.round(seconds) + 's';
+          } else {
+              duration = (Math.round(seconds * 100) / 100) + 's';
+          }
+          return duration;
+
+        }},
+
+        getUserJobs: {
+          value : function(){
+            var self = this;
+
+            var seconds = (new Date().getTime() / 1000) - 172800;
+
+            var now  = self.nowDate  || (new Date()).getTime();
+            var then = self.thenDate || now - self.numHours * 60 * 60 * 1000;
+
+            var userJobs = [];
+
+            var token = this.runtime.service('session').getAuthToken();
+            this.metricsClient = new DynamicService({
+              url: this.runtime.getConfig('services.service_wizard.url'),
+              token: token,
+              version : 'dev',
```

Comment:
Unless there is a beta or release in the CI catalog, can't you omit the version -- it will default to the most "released" version, which will be "dev".
Note to self, though: we need, and have discussed before, the ability to control the version for a module in the ui runtime...